tools/sense_studio/sense_studio.py

```diff
@@ -391,6 +399,85 @@ def set_timer_default():
+        counterpart_tags_path = os.path.join(path, f'tags_{split}', counterpart_class_name)
+
+        # Create directory to save flipped videos
+        os.makedirs(videos_path_out, exist_ok=True)
+        os.makedirs(counterpart_tags_path, exist_ok=True)
+
+        video_list = [video for video in os.listdir(videos_path_in) if video.endswith('.mp4')]
+
+        for video in video_list:
+            output_video_list = [op_video for op_video in os.listdir(videos_path_out) if op_video.endswith('.mp4')]
+            print(f'Processing video: {video}')
+            if '_flipped' in video:
+                flipped_video_name = ''.join(video.split('_flipped'))
+            else:
+                flipped_video_name = video.split('.')[0] + '_flipped' + VIDEO_EXT
+
+            if flipped_video_name not in output_video_list:
+                # Original video as input
+                original_video = ffmpeg.input(os.path.join(videos_path_in, video))
+                # Do horizontal flip
+                flipped_video = ffmpeg.hflip(original_video)
+                # Get flipped video output
+                flipped_video_output = ffmpeg.output(flipped_video,
+                                                     filename=os.path.join(videos_path_out, flipped_video_name))
+                # Run to render and save video
+                ffmpeg.run(flipped_video_output)
+
+                # Copy tags of original video to flipped video (in train set)
+                if video in copy_train_video_tags and split == 'train':
+                    original_tags_file = os.path.join(original_tags_path, video.replace('.mp4', '.json'))
+                    flipped_tags_file = os.path.join(counterpart_tags_path, flipped_video_name.replace('.mp4', '.json'))
+
+                    if os.path.exists(original_tags_file):
+                        with open(original_tags_file) as f:
+                            original_video_tags = json.load(f)
+                        original_video_tags['file'] = flipped_video_name
+                        with open(flipped_tags_file, 'w') as f:
+                            f.write(json.dumps(original_video_tags, indent=2))
+
+                # Copy tags of original video to flipped video (in valid set)
+                if video in copy_valid_video_tags and split == 'valid':
+                    original_tags_file = os.path.join(original_tags_path, video.replace('.mp4', '.json'))
+                    flipped_tags_file = os.path.join(counterpart_tags_path, flipped_video_name.replace('.mp4', '.json'))
+
+                    if os.path.exists(original_tags_file):
+                        with open(original_tags_file) as f:
+                            original_video_tags = json.load(f)
+                        original_video_tags['file'] = flipped_video_name
+                        with open(flipped_tags_file, 'w') as f:
+                            f.write(json.dumps(original_video_tags, indent=2))
```

Comment:
Is there a difference between the "train" and "valid" cases? They look identical to me. Might be missing something

Reply:
There are two lists coming from the UI separately for train and valid, as there might be a case where both may have same named video

Reply:
But I'll think something if I can merge this